GoogleSignIn/Tests/Unit/GIDSignInTest.m

```diff
@@ -738,136 +742,185 @@ - (void)testNotHandleWrongPath {
+  GIDHTTPFetcherTestBlock testBlock =
+      ^(NSURLRequest *request, GIDHTTPFetcherFakeResponseHandlerBlock response) {
+        NSURL *url = request.URL;
+        XCTAssertEqualObjects([url scheme], @"https", @"scheme must match");
+        XCTAssertEqualObjects([url host], @"accounts.google.com", @"host must match");
+        XCTAssertEqualObjects([url path], @"/o/oauth2/revoke", @"path must match");
+        OIDURLQueryComponent *queryComponent = [[OIDURLQueryComponent alloc] initWithURL:url];
+        NSDictionary<NSString *, NSObject<NSCopying> *> *params = queryComponent.dictionaryValue;
+        XCTAssertEqualObjects([params valueForKey:kSDKVersionLoggingParameter], GIDVersion(),
+                              @"SDK version logging parameter should match");
+        XCTAssertEqualObjects([params valueForKey:kEnvironmentLoggingParameter], GIDEnvironment(),
+                              @"Environment logging parameter should match");
+        
+        NSData *body = request.HTTPBody;
+        NSString* bodyString = [[NSString alloc] initWithData:body encoding:NSUTF8StringEncoding];
+        NSArray<NSString *> *strings = [bodyString componentsSeparatedByString:@"="];
+        XCTAssertEqualObjects(strings[1], kAccessToken);
+
+        NSData *data = [[NSData alloc] init];
+        response(data, nil);
+        [expectation1 fulfill];
+     };
+  [_httpFetcher setTestBlock:testBlock];
+  
+  XCTestExpectation *expectation =
+      [self expectationWithDescription:@"Callback called with nil error"];
+  [_signIn disconnectWithCompletion:^(NSError * _Nullable error) {
+    XCTAssertNil(error);
+    [expectation fulfill];
+  }];
+  [self waitForExpectationsWithTimeout:1 handler:nil];
+  XCTAssertNil([_keychainHandler loadAuthState]);
+}
+
 // Verifies disconnect calls callback with no errors if access token is present.
-- (void)testDisconnect_accessToken {
+- (void)testDisconnect_accessTokenIsPresent {
   [_keychainHandler saveAuthState:_authState];
-  [[[_authState expect] andReturn:_tokenResponse] lastTokenResponse];
-  [[[_tokenResponse expect] andReturn:kAccessToken] accessToken];
-  [[[_authorization expect] andReturn:_fetcherService] fetcherService];
+  OCMStub([_authState lastTokenResponse]).andReturn(_tokenResponse);
+  OCMStub([_tokenResponse accessToken]).andReturn(kAccessToken);
+  
+  GIDHTTPFetcherTestBlock testBlock =
+      ^(NSURLRequest *request, GIDHTTPFetcherFakeResponseHandlerBlock responseHandler) {
+        NSData *data = [[NSData alloc] init];
+        responseHandler(data, nil);
+     };
+  [_httpFetcher setTestBlock:testBlock];
+  
   XCTestExpectation *expectation =
       [self expectationWithDescription:@"Callback called with nil error"];
   [_signIn disconnectWithCompletion:^(NSError * _Nullable error) {
-    if (error == nil) {
-      [expectation fulfill];
-    }
+    XCTAssertNil(error);
+    [expectation fulfill];
   }];
-  [self verifyAndRevokeToken:kAccessToken hasCallback:YES];
```

Comment:
The test case used to verify the token is revoked. Please add it back. You can reuse the verification code in `testDisconnect_verifyURL_succeed`.